### PR TITLE
Added `Custom` derive macro

### LINKED ISSUE #9: Add a no-op derive macro that brings attribute "custom" into scope
Motivation:

```rust
use macro_rules_attribute::derive;

#[derive(MyCustomDerive!)]
#[my_custom_derive(foo = "bar")]
struct Foo;
```

This code example is erroneous, since `my_custom_derive` is not in the scope.

Proposed solution:

```rust
use macro_rules_attribute::{derive, CustomAttr};

#[derive(MyCustomDerive!, CustomAttr)]
#[custom(my_custom_derive(foo = "bar"))]
struct Foo;
```

`CustomAttr` is a no-op derive macro that produces an empty token stream, but brings attribute `custom` into scope. `macro_rules` derive can then use said attribute.

### PR DESCRIPTION
**Pull request by ModProg**
This allows using `#[custom(...)]` in macro_rules derive

#9

### CODE REVIEW

**Comment by ModProg** [comment]
One thing I considered, but wasn't sure about, is deriving it by default with `macro_rules_derive`.

**Comment by danielhenrymantilla** [comment]
Nice idea (apologies for the late review); for what is worth, this is definitely mergeable, but I share your sentiment about directly integrating this into `macro_rules_attribute`.

The issue at that point would be that `custom` seems a tad too generic to add like this (potential collision with a genuine `#[custom]` macro)🤔

  - what about `#[derive_args(...)]`?

Another option would be our `#[derive()]`, once it has detected the custom `Macro!`s, to snake-case them (_e.g._, `#[macro]`) and make those inert.

That way:

```rs
#[derive(MyCustomDerive!)]
#[my_custom_derive(foo = "bar")]
```

would Just Work™

**Review by danielhenrymantilla — APPROVED** [review]

File: src/lib.rs [hunk 1]
```diff
@@ -518,6 +518,11 @@ fn main ()
 /// procedural macro one or a `macro_rules!` one.
 pub use ::macro_rules_attribute_proc_macro::derive;
 
+
+/// No-op macro that can be used with [`derive`] or [`macro_rules_derive`] to allow
+/// using the `#[custom(...)]` attribute.
+pub use ::macro_rules_attribute_proc_macro::Custom;
```
Comment (danielhenrymantilla) [inline src/lib.rs]:
```suggestion
/// using the `#[custom(...)]` attribute.
///
/// See https://github.com/danielhenrymantilla/macro_rules_attribute-rs/issues/9 for more info.
pub use ::macro_rules_attribute_proc_macro::Custom;
```

**Review by danielhenrymantilla — APPROVED** [review]
I think we can merge this with the added doc-comment (feel free to also run `./change_version 0.1.4` too in a separate `Version bump` commit so that I can directly release it); I'll look into my own more fine-tuned proposal when I have the time 🙂

**Comment by ModProg** [comment]
> Another option would be our `#[derive()]`, once it has detected the custom `Macro!`s, to snake-case them (_e.g._, `#[macro]`) and make those inert.

The problem with that is, that IIRC there is no way to dynamically support custom attributes, this means, we would need to remove them manually from the struct, possible, but we need to be careful then to not remove ones that another derive macro needs.



> The issue at that point would be that `custom` seems a tad too generic to add like this (potential collision with a genuine `#[custom]` macro)thinking
> 
>     * what about `#[derive_args(...)]`?

We could also support multiple so that a user can just use the one they prefer/doesn't conflict, as rust doesn't have an issue with multiple derive macros sharing helper attributes.

**Comment by ModProg** [comment]
@danielhenrymantilla 
> We could also support multiple so that a user can just use the one they prefer/doesn't conflict, as rust doesn't have an issue with multiple derive macros sharing helper attributes.

If we go with that, what would be your preferred names? And should we derive it by default? We could put it behind an optional feature in case you don't want the probably very negligible overhead of deriving a macro, that does nothing in cases where it isn't used.

**Comment by danielhenrymantilla** [comment]
[hmm, I had missed the notification]

I guess `#[custom]` and `#[derive_args]` could be a baseline before thinking about others / I think those two would suffice 🙂 

Anyhow, I'm definitely tempted to merge this already as is, but I'll let you add anything last-minute you may wish

  - (don't mind that CI thing, it's a github runner regression which I need to cater to at some point)

**Comment by ModProg** [comment]
> I guess `#[custom]` and `#[derive_args]` could be a baseline before thinking about others / I think those two would suffice slightly_smiling_face

Added `derive_args`.

What about deriving it by default?

**Comment by danielhenrymantilla** [comment]
Yeah, let's do it. I don't think the performance penalty will be noticeable, and it will be ergonomic 👍

**Comment by ModProg** [comment]
> Yeah, let's do it. I don't think the performance penalty will be noticeable, and it will be ergonomic

yeah, it would only be a very small compile time hit. 

@danielhenrymantilla  This should now be ready for merge from my side.

**Comment by danielhenrymantilla** [comment]
Thanks @ModProg !

**Pull request merged**